Makefile

```
@@ -41,6 +41,9 @@ lint: venv
 	$(VENVDIR)/bin/python3 -m pre_commit --version > /dev/null || $(VENVDIR)/bin/python3 -m pip install pre-commit
 	$(VENVDIR)/bin/python3 -m pre_commit run --all-files
 
+test: venv
+	$(VENVDIR)/bin/python3 -m pytest --cov pep_sphinx_extensions --cov-report html --cov-report term --cov-report xml
```

Instead of glomming this config into the command line invocation in this particular instance, can we put it in the `addopts` in a `pytest.ini` instead, so its in a standard location in a declarative format, and it gets picked up consistently by any Pytest invocations rather than just this one run through the platform-specific `make`?

And while we're at, why not use more robust and informative defaults? 

So I'd suggest a `pytest.ini` like this:

```ini
addopts = -v -r a --color=yes --code-highlight=yes --strict-config --strict-markers --import-mode=importlib--cov pep_sphinx_extensions --cov-report html --cov-report term --cov-report xml
empty_parameter_set_mark = fail_at_collect
filterwarnings =
    error
minversion = 6.0
xfail_strict = True
```

Also, it couldn't hurt being a bit more rigorous here as well...

```suggestion
	$(VENVDIR)/bin/python3 -bb -X dev -W error -m pytest
```

---

I actually noticed another issue in my local testing—the entire project tree is getting searched for tests, which not only is rather inefficient but it means files named `test_*.py` in the PEP subdirectories, e.g. `pep-0446\test_cloexec.py`, are getting picked up. This actually leads to a crash on collection when I tested locally on my Windows machine, even though that particular file contained no functions named `test_*` that Pytest runs at test time, because it contains `import fnctl` which isn't available on non-Unix platforms (the same would be true of any file that contains invalid code or missing dependencies).

Adding `testpaths = pep_sphinx_extensions` fixes this, ensuring that only tests within the PEP build system tree are picked up. I've updated the proposed `pytest.ini` with that fix, and the other elisions mentioned above. I've also tested it locally with such file, and all the tests pass and it works great.

>  I don't want to see the bug multiline --cov-report term output.

Okay, I elided it from there and we can pass it in the invocation that feeds the CIs.

> -v: no need to print all the test names by default.

Personally, I find the non-verbose test status output cryptic and most if not almost all the various other projects I've been involved with use `-v` or `-vv` as standard (at least on CI). It also does more than that, particularly in terms of printing more detailed diffs and not truncating them when failures occur which could hide key information, particularly again on CI, where users can't simply re-run with `-v` (unless they set up a test env locally and then run it themselves).

Since this is mostly a problem on CI, where users can't pass their own `-v` flags and where verbosity is less of a problem and more of an asset, what about dropping `-v` in the `pytest.ini` file, but just adding it in the GHA/CI-level invocation? This would ensure CI output is more complete, while allowing not doing so by default locally. I dropped it accordingly from the above, as requested.

> `--color=yes`: the default `auto` shows colour fine in my terminal, and we have the `FORCE_COLOR: 1` env var as a catch all for all tools in the CI. Does `auto` work for Windows?

Yeah, it does—this is in my standard `pytest.ini` to ensure it was shown on CI, and I didn't realize we had `FORCE_COLOR: 1` for that already. Therefore, I elided it as requested.

> `--code-highlight=yes`: this is already the default

Ah, okay. The default wasn't documented anywhere I could easily find searching the Pytest docs, so I wasn't sure. Removed.

> Please could you explain `--import-mode=importlib`?

Sure, see the [Pytest docs on the subject](https://docs.pytest.org/en/7.1.x/explanation/pythonpath.html?highlight=importlib). Since it is the recommended mode, much more robust than the hacky legacy `prepend`/`append` and will become the default in future releases, may as well adopt it it now.

---

Thanks! Let's go with this (and will add `pytest.ini`).